src/_arraykit.c

```diff
@@ -491,6 +491,318 @@ isna_element(PyObject *Py_UNUSED(m), PyObject *arg)
     Py_RETURN_FALSE;
 }
 
+//------------------------------------------------------------------------------
+// get_new_indexers_and_screen & related
+
+static PyObject *
+AK_get_index_screen(PyObject *element_locations,
+                    PyObject *positions,
+                    PyObject *num_unique
+                    )
+{
+    /*
+    Returns the index screen built up from an array of element locations and a
+    positions array. The index screen will be applied to the original index to
+    determine the labels of what the new indexers map to
+
+    num_unique == len(positions), and is used as the flag to determine which
+    elements were not found.
+
+    Also, len(element_locations) == len(positions).
+
+    Example:
+        input:
+            element_locations = [6, 3, 1, 6, 0, 2]
+            positions         = [0, 1, 2, 3, 4, 5]
+            num_unique        = 6
+
+        algorithm:
+            found_mask        = [F, T, T, F, T, T] (element_locations != num_unique)
+            found_locations   = [3, 1, 0, 2]       (element_locations[found_mask])
+            found_positions   = [1, 2, 4, 5]       (positions[found_mask])
+
+        output:
+                                [4, 2, 5, 1]       (found_positions[argsort(found_locations)])
+
+    Equivalent Python code:
+
+        found_mask = element_locations != num_unique             # 1
+
+        found_element_locations = element_locations[found_mask]  # 2
+        order_found = np.argsort(found_element_locations)        # 3
+
+        found_positions = positions[found_mask]                  # 4
+        return found_positions[order_found]                      # 5
+    */
+    // 1. Build mask
+    PyObject *found_mask = PyObject_RichCompare(element_locations, num_unique, Py_NE);
+    if (found_mask == NULL) {
+        return NULL;
+    }
+
+    // 2. Apply mask to ``element_locations``
+    PyObject *found_element_locations = PyObject_GetItem(element_locations, found_mask);
+    if (found_element_locations == NULL) {
+        Py_DECREF(found_mask);
+        return NULL;
+    }
+
+    // 3. Argsort ``found_element_locations``
+    // Quicksort is safe since array just contains integers
+    PyObject* order_found = PyArray_ArgSort(
+            (PyArrayObject*)found_element_locations, // array
+            0,                                       // axis
+            NPY_QUICKSORT                            // kind
```

Comment:
I do not see the availability of this argument in the public API docs!
https://numpy.org/doc/stable/reference/c-api/array.html#c.PyArray_ArgSort

Reply:
That is very strange! I see the argument defined here, and everything is worked.

https://github.com/numpy/numpy/blob/main/numpy/core/src/multiarray/item_selection.c#L1419